### ISSUE TITLE
Use OS native line ending for "insert_final_newline" check when "end_of_line = unset"

### ISSUE THREAD

**Issue by Ollie-Moss2** [editorconfig-checker/editorconfig-checker#580]
## Overview:
The "insert_final_newline" check should default to using the OS native line ending when checking for a final new line when "end_of_line = unset".

## Current Behaviour:
Given this file and editorconfig I get an unexpected error:\
\
**myjsonfile.json:**
```json
{
   "somejson": "with a final crlf final new line",
}
\r\n
```

**Editor Config:**
```ini
# All files
[*]
indent_style = space
# Line endings are handled because we use text=auto in gitattributes 
end_of_line = unset
indent_size = 2
insert_final_newline = true
```

**Output:**
```
myjsonfile.json:
        Wrong line endings or no final newline
```

## Expected Behaviour:
No error to be produced.

## Proposed Fix:
Update `utils.go:GetEolChar()` to default to OS native line ending and not just `\n`.\
*I don't know Go so I am unsure exactly how this would be implemented*
```go
func GetEolChar(endOfLine string) string {
	switch endOfLine {
	case "lf":
		return "\n"

	case "cr":
		return "\r"
	case "crlf":
		return "\r\n"
	}

    // default to OS native line ending here...
        
	return "\n"
}
```

**Comment by mstruebing** [comment]
I'm not 100% sure right now but isn't it implemented that when it's unset it doesn't fulfill the check at all?
If not, I think this should be the right behavior.
